guides/https.md

```diff
@@ -0,0 +1,255 @@
+
+```elixir
+Plug.Cowboy.https MyApp.MyPlug, [],
+  port: 8443,
+  cipher_suite: :strong,
+  certfile: "/etc/letsencrypt/live/example.net/cert.pem",
+  keyfile: "/etc/letsencrypt/live/example.net/privkey.pem",
+  cacertfile: "/etc/letsencrypt/live/example.net/chain.pem"
+```
+
+The `cacertfile` option is not needed when using a self-signed certificate, or when the file pointed to by `certfile` contains both the server certificate and all necessary CA chain certificates:
+
+```elixir
+  #...
+  certfile: "/etc/letsencrypt/live/example.net/fullchain.pem",
+  keyfile: "/etc/letsencrypt/live/example.net/privkey.pem"
+```
+
+It is possible to bundle the certificate files with the  application, possibly for packaging into a release. In this case the files must be stored under the application's 'priv' directory. The `otp_app` option must be set to the name of the OTP application that contains the files, in order to correctly resolve the relative paths:
+
+```elixir
+Plug.Cowboy.https MyApp.MyPlug, [],
+  port: 8443,
+  cipher_suite: :strong,
+  certfile: "priv/cert/selfsigned.pem",
+  keyfile: "priv/cert/selfsigned_key.pem",
+  otp_app: :my_app
+```
+
+Remember to exclude the files from version control, unless the certificate and key are shared by all developers for testing purposes only. For example, add this line to the '.gitignore' file: `priv/**/*.pem`.
+
+## TLS Protocol Options
+
+In addition to a certificate, an HTTPS server needs a secure TLS protocol configuration. `Plug.SSL` always sets the following options:
+
+* Set `secure_renegotiate: true`, to avoid certain types of man-in-the-middle attacks
+* Set `reuse_sessions: true`, for improved handshake performance of recurring connections
+
+Additional options can be set by selecting a predefined profile or by setting `:ssl` options individually.
+
+### Predefined Options
+
+To simplify configuration of TLS defaults Plug provides two preconfigured options: `cipher_suite: :strong` and `cipher_suite: :compatible`.
+
+The `:strong` profile enables AES-GCM ciphers with ECDHE or DHE key exchange, and TLS version 1.2 only. It is intended for typical installations with support for browsers and other modern clients.
+
+The `:compatible` profile additionally enables AES-CBC ciphers, as well as TLS versions 1.1 and 1.0. Use this configuration to allow connections from older clients, such as older PC or mobile operating systems. Note that RSA key exchange is not enabled by this configuration, due to known weaknesses, so to support clients that do not support ECDHE or DHE it is necessary specify the ciphers explicitly (see [below](#manual-configuration)).
+
+In addition, both profiles:
+
+* Configure the server to choose a cipher based on its own preferences rather than the client's (`honor_cipher_order` set to `true`); when specifying a custom cipher list, ensure the ciphers are listed in descending order of preference
+* Set `client_renegotiation: false`
```

Comment:
We may want to reconsider this (in the implementation): renegotiation may be necessary to reset sequence numbers on long-lived connections, and if rejected by the server will cause such connections to close. To the best of my knowledge `:ssl` rate-limits client renegotiation requests to minimize the risk of DoS.

Reply:
I'll create a separate PR for this, so we can discuss it with the relevant people